Mathlib/NumberTheory/ModularForms/Basic.lean

```diff
@@ -430,3 +437,29 @@ example (Γ : Subgroup SL(2, ℤ)) : Algebra ℂ (⨁ i, ModularForm Γ i) := in
 end GradedRing
 
 end ModularForm
+
+section translate
+open ModularForm
+
+variable {k : ℤ} {Γ : Subgroup SL(2, ℤ)} {F : Type*} [FunLike F ℍ ℂ] (f : F) (g : SL(2, ℤ))
```

Comment:
How much more work is to have `g` be an element of `GL(2,ℚ)⁺`  and then map into `Γ ∩ g Γ g⁻¹` since this is what we'll eventually want (at least) for Hecke operators right?

Also maybe we call "g" something like "A" instead in the theorems? Since I think in this namespace we use `f,g` for modular forms and `A,B` for matrices.

Reply:
> How much more work is to have `g` be an element of `GL(2,ℚ)⁺` and then map into `Γ ∩ g Γ g⁻¹` since this is what we'll eventually want (at least) for Hecke operators right?

Quite a lot more difficult I'm afraid: the difficulty is showing that the `translate`d form is bounded at cusps. I thought about this for a while and realised that the easiest way to do it would probably be to write a general element of `GL(2,ℚ)⁺` as a product of elts of `SL2Z` and diagonal matrices, so there is some value in pursuing the SL2Z case separately first.

Reply:
Fair enough, I guess we can add a TODO to say we'll fix this once we add Hecke operators.

Reply:
I added the TODO notes. I don't like using `A` for the variable; using `g` for a generic element of SL2 or GL2 seems natural to me and is also a well-established convention in the `ModularGroup` code.